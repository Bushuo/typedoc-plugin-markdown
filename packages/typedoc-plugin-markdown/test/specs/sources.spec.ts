import * as Handlebars from 'handlebars';
import { DeclarationReflection } from 'typedoc';

import { TestApp } from '../test-app';

const getProp = (reflection: DeclarationReflection) => {
  const prop = reflection.findReflectionByName('prop');
  prop.sources = prop.sources.map((source) => {
    delete source.url;
    return source;
  });
  return prop;
};

describe(`Sources:`, () => {
  let testApp: TestApp;
  let partial: Handlebars.TemplateDelegate;
  beforeAll(async () => {
    testApp = new TestApp(['sources.ts']);
    await testApp.bootstrap();
    partial = TestApp.getPartial('member.sources');
  });

  test(`should display implementation of sources'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        getProp(testApp.findReflection('SomeClass')),
      ),
    ).toMatchSnapshot();
  });

  test(`should display inherited sources'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        getProp(testApp.findReflection('AnotherInterface')),
      ),
    ).toMatchSnapshot();
  });

  test(`should display overide sources'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        getProp(testApp.findReflection('AnotherClass')),
      ),
    ).toMatchSnapshot();
  });
});
