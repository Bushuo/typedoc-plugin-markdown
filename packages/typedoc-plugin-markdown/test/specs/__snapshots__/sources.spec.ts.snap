// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sources: should display implementation of sources' 1`] = `
"#### Implementation of

[SomeInterface](someinterface.md).[prop](someinterface.md#prop)

#### Defined in

sources.ts:6
"
`;

exports[`Sources: should display inherited sources' 1`] = `
"#### Inherited from

[SomeClass](../classes/someclass.md).[prop](../classes/someclass.md#prop)

#### Defined in

sources.ts:2
"
`;

exports[`Sources: should display overide sources' 1`] = `
"#### Overrides

[SomeClass](../classes/someclass.md).[prop](../classes/someclass.md#prop)

#### Defined in

sources.ts:10
"
`;
